docs/general-project-conventions.md

```diff
@@ -55,29 +55,45 @@ export
 
 Services that need to communicate with services in other projects should join the `COMMON_NET_NAME` network. External ports may also be mapped if needed.
 
 This layer should not depend on any files outside of the root directory of the repo (to allow for shallow cloning).
 
-### Local Development
+### "Build"
 
-This layer builds on the above by adding the necessary build commands to the docker compose services. This layer would typically be used by the `make build` target.
+This layer includes the minimum that is required to build the project and run unit tests. It should not depend heavily on other services, so is likely to extend the Raw layer rather than the Run layer. This layer would typically be used by the `make build` target.
 
-### Production Deployment
+### "Test"
 
-The layer would extend the Local Runtime layer with any production-specific configurations.
+This layer extends the Build layer with any additional services that are required to run integration tests. Note that this should not extend the Run layer as this would then create a dependency on external services with persisted state. If external services are needed then they should be included using the external project's Raw layer to create a minimal container with transient state.
+
+Any external services introduced in this layer would be incorporated into the current project's Docker namespace.
+
+### Local "Dev"
+
+This layer is a combination of the Build and Run layer that allows a developer to run and debug the service locally, pulling in other services as dependencies. This should be the default environment for docker commands so that these can be used without having to rely on equivalents in the Makefile. 
+
+### "Prod" Deployment
+
+The layer would extend the Raw layer with production-specific configurations. It should not reference any dependent or shared services as these will have their own production deployment scripts.
+
+### Diagram
+
+The following diagram depicts these layers in a class style, with the make commands shown as methods:
+
+[![](https://mermaid.ink/img/pako:eNp1Uk1PwzAM_SuRTyBtf6DihHbhgITGTigSCom7BbVOlY8hNPrfcZKNtQV8cfz0_J6d5ATaGYQGdKdC2Fi196qXJDgKIrbqQ9IMSCROFcixxRY9ksYgXDygF4N376hjIV5pPtHNbS3HmlhY3H2t15U3dbhPtjNTj7cM_Nde2HOBHYY47X8g3SXzx4R5twvJUkRePlpHr5EFFn51qOKY5eeGGzxO_QweL9050rCcna-wKHHfL_WCTcWfvDOLnSsEK-jR98oafr9iL4EX7FFCw0eDrUpdlCBpZKpK0T1_koYm-oQr8C7tD9C0qgtcpcGoiOf3_0EHRS_OXWs0Njr_eP4xOY3fPD6xSg?type=png)](https://mermaid.live/edit#pako:eNp1Uk1PwzAM_SuRTyBtf6DihHbhgITGTigSCom7BbVOlY8hNPrfcZKNtQV8cfz0_J6d5ATaGYQGdKdC2Fi196qXJDgKIrbqQ9IMSCROFcixxRY9ksYgXDygF4N376hjIV5pPtHNbS3HmlhY3H2t15U3dbhPtjNTj7cM_Nde2HOBHYY47X8g3SXzx4R5twvJUkRePlpHr5EFFn51qOKY5eeGGzxO_QweL9050rCcna-wKHHfL_WCTcWfvDOLnSsEK-jR98oafr9iL4EX7FFCw0eDrUpdlCBpZKpK0T1_koYm-oQr8C7tD9C0qgtcpcGoiOf3_0EHRS_OXWs0Njr_eP4xOY3fPD6xSg)
```

Comment:
dependencies should also have a "stop" method, because it's a bit tedious to stop them otherwise (so on parent project we can have run_deps and stop_deps).

Reply:
OK. I think that's probably a separate discussion as it would be more than a cosmetic implementation